### PR TITLE
Parse USBmon Header

### PR DESCRIPTION
**Pull request by Swiftb0y**
Currently, the usb layer can only parse USBpcap headers and is thus windows only. The goal of this PR is to bring USBmon parsing capabilities so scapy

It's far from done, but I need some help along the way, which is why I'm opening the PR now.

TODO:
 - [ ] implement sniffing
 - [ ] Add auto recognition of packet in stream/pcap
 - [ ] create tests
 - [ ] resolve issues described in comments

### CODE REVIEW

File: scapy/layers/usb.py [hunk 1]
```diff
@@ -121,6 +122,86 @@ def guess_payload_class(self, payload):
...
+    def guess_payload_class(self, payload):
+        if self.xfer_type == 0:
+            return USBmonTransferIsochronous
+        elif self.xfer_type == 1:
+            return USBmonTransferInterrupt
+        elif self.xfer_type == 2:
+            return USBmonTransferControl
+        elif self.xfer_type == 3:
+            return USBmonTransferBulk
+        return conf.raw_layer
```
Comment (gpotter2) [inline scapy/layers/usb.py:186]: You don't need this if you use the bindings

**Comment by Swiftb0y** [comment]
I'm sorry, but I switched away from scapy due to the lack of documentation and I won't put any more work into this PR.

**Comment by gpotter2** [comment]
Thanks for your work on this PR.
It's alright. (though i find it quite offensing to say there's [no doc...](https://scapy.readthedocs.io/en/latest/)).

Have a great day

**Comment by Swiftb0y** [comment]
I'm sorry. To clarify, I didn't mean no doc, there is (as you linked) and it helped me to understand the concepts behind scapy, but neither the doc nor the source documents the API to the  degree of detail I would like. To understand scapy to the point where I could integrate everything I need, I would have to read all the implementation details of every single field. That is just too time consuming for something that wiil sooner or later have to be rewritten in JS anyways (because it is actually made for a [different project](https://github.com/mixxxdj/mixxx).

**Pull request closed without merging**